test/quantization/test_backward_compatibility.py

```diff
@@ -67,76 +68,55 @@ def remove_prefix(text, prefix):
         self.assertEqual(qmodule_scripted(input_tensor), expected, atol=prec)
         self.assertEqual(qmodule_traced(input_tensor), expected, atol=prec)
 
-    # TODO: add qnnpack test(https://github.com/pytorch/pytorch/pull/36771)
-    @unittest.skipUnless(
-        'fbgemm' in torch.backends.quantized.supported_engines,
-        " Quantized operations require FBGEMM. FBGEMM is only optimized for CPUs"
-        " with instruction set support avx2 or newer.",
-    )
     def test_linear(self):
-        module = nnq.Linear(3, 1, bias_=True, dtype=torch.qint8)
-        self._test_op(module, input_size=[1, 3], generate=False)
-
-    @unittest.skipUnless(
-        'fbgemm' in torch.backends.quantized.supported_engines,
-        " Quantized operations require FBGEMM. FBGEMM is only optimized for CPUs"
-        " with instruction set support avx2 or newer.",
-    )
+        for i, qengine in enumerate(supported_qengines):
+            with override_quantized_engine(qengine):
```

Comment:
optional: thoughts on using a decorator for `with override_quantized_engine(qengine)`, so we can save a level of indent for the test cases?  something like below, and it could wrap the "exit if not supported" logic too

```
@override_qengines
def test_foo(...):
    ...
```

Reply:
I did think about it, but wasn't sure how it would play with the hypothesis params we have and if it would make errors harder to reproduce. Let me try it out